GCRCatalogs/dc2_coadd.py

```diff
@@ -34,91 +88,291 @@ def _subclass_init(self, **kwargs):
 
-        self._quantity_modifiers = {
+        self._quantity_modifiers = self._generate_modifiers()
+        self._datasets, self._columns = self._generate_datasets_and_columns()
+        if not self._datasets:
+            err_msg = 'No catalogs were found in `base_dir` {}'
+            raise RuntimeError(err_msg.format(self._base_dir))
+
+    @staticmethod
+    def _generate_modifiers():
+        """Creates a dictionary relating native and homogenized column names
+
+        Returns:
+            A dictionary of the form {<homogenized name>: <native name>, ...}
+        """
+        modifiers = {
+            'objectId': 'id',
+            'parentObjectId': 'parent',
             'ra': (np.rad2deg, 'coord_ra'),
             'dec': (np.rad2deg, 'coord_dec'),
+            'centroidX': 'slot_Centroid_x',
+            'centroidY': 'slot_Centroid_y',
+            'centroidX_err': 'slot_Centroid_xSigma',
+            'centroidY_err': 'slot_Centroid_ySigma',
+            'centroid_flag': 'slot_Centroid_flag',
+            'psNdata': 'base_PsfFlux_area',
+            'extendedness': 'base_ClassificationExtendedness_value',
         }
 
+        modifiers['good'] = (
+            create_basic_flag_mask,
+            'deblend_skipped',
+            'base_PixelFlags_flag_edge',
+            'base_PixelFlags_flag_interpolatedCenter',
+            'base_PixelFlags_flag_saturatedCenter',
+            'base_PixelFlags_flag_crCenter',
+            'base_PixelFlags_flag_bad',
+            'base_PixelFlags_flag_suspectCenter',
+            'base_PixelFlags_flag_clipped',
+        )
+
+        # cross-band average, second moment values
+        modifiers['I_flag'] = 'slot_Shape_flag'
+        for ax in ['xx', 'yy', 'xy']:
+            modifiers['I{}'.format(ax)] = 'slot_Shape_{}'.format(ax)
+            modifiers['I{}PSF'.format(ax)] = 'slot_PsfShape_{}'.format(ax)
+
         for band in 'ugrizy':
-            self._quantity_modifiers['mag_{}_lsst'.format(band)] = '{}_mag'.format(band.lower())
-            self._quantity_modifiers['magerr_{}_lsst'.format(band)] = '{}_mag_err'.format(band.lower())
+            modifiers['mag_{}_lsst'.format(band)] = '{}_mag'.format(band)
+            modifiers['magerr_{}_lsst'.format(band)] = '{}_mag_err'.format(band)
+            modifiers['{}_psFlux'.format(band)] = '{}_slot_ModelFlux_flux'.format(band)
+            modifiers['{}_psFlux_flag'.format(band)] = '{}_slot_ModelFlux_flag'.format(band)
+            modifiers['{}_psFlux_err'.format(band)] = '{}_slot_ModelFlux_fluxSigma'.format(band)
+
+            # Band specific second moment values
+            modifiers['{}_I_flag'.format(band)] = '{}_slot_Shape_flag'.format(band)
+
+            for ax in ['xx', 'yy', 'xy']:
+                modifiers['{}_I{}'.format(band, ax)] = '{}_slot_Shape_{}'.format(band, ax)
+                modifiers['{}_I{}PSF'.format(band, ax)] = '{}_slot_PsfShape_{}'.format(band, ax)
+
+            modifiers['{}_mag_CModel'.format(band)] = (
```

Comment:
Since this is not part of DPDD, we can rename it to `mag_{}_cmodel` so that it follows the same convention as `mag_{}_lsst`, @EiffL?